### PR TITLE
Viewport improvements

### PR DESCRIPTION
**Pull request by Shqrdx**
Changes : 
  -Comments for all things inside Viewport struct.
  -New constructor which accept full Viewport description.
  -Change TitleSafeArea a bit.(I still thinks that is subject to change for platforms like PS4 or XBOX)

PS If this change is not correct I remove it quickly.

### CODE REVIEW

File: MonoGame.Framework/Graphics/Viewport.cs [hunk 1]
```diff
@@ -1,55 +1,18 @@
...
-code form, you may only do so under a license that complies with this license.
-(E) The software is licensed "as-is." You bear the risk of using it. The contributors give no express warranties, guarantees
-or conditions. You may have additional consumer rights under your local laws which this license cannot change. To the extent
-permitted under your local laws, the contributors exclude the implied warranties of merchantability, fitness for a particular
-purpose and non-infringement.
-*/
-#endregion License
-
-using Microsoft.Xna.Framework;
+// MIT License - Copyright (C) The Mono.Xna Team
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]: This is the wrong license header.  It should be the MonoGame one.

File: MonoGame.Framework/Graphics/Viewport.cs [hunk 2]
```diff
@@ -58,6 +21,10 @@ public struct Viewport
 		private float maxDepth;
 		
 		#region Properties
+
+        /// <summary>
+        /// The height of this <see cref="Viewport"/>.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]: We don't need to include `of this <see cref="Viewport"/>` on every summary.  Also the link is redundant as it is just linking to the same docs they are already reading.

Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]: For example a more useful description could be...
> The height of the bounds in pixels.

**Comment by Shqrdx** [comment]
@tomspilman Whats I must change next ?

File: MonoGame.Framework/Graphics/Viewport.cs [hunk 3]
```diff
@@ -172,11 +173,42 @@ public Viewport(int x, int y, int width, int height)
 		    this.minDepth = 0.0f;
 		    this.maxDepth = 1.0f;
 		}
-		
+
+        /// <summary>
+        /// Creates a new instance of <see cref="Viewport"/> struct.
+        /// </summary>
+        /// <param name="x">The x coordinate of the upper-left corner, of the view bounds, in pixels.</param>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]: The commas are not necessary here.

Also use this same description for the X/Y property as well.

File: MonoGame.Framework/Graphics/Viewport.cs [hunk 4]
```diff
@@ -154,15 +138,32 @@ public Rectangle Bounds
...
 			}
 		}
 		
+        /// <summary>
+        /// Creates a new instance of <see cref="Viewport"/> struct.
+        /// </summary>
+        /// <param name="x">The x coordinate of the view bounds.</param>
+        /// <param name="y">The y coordinate of the view bounds.</param>
+        /// <param name="width">The width of the view bounds.</param>
+        /// <param name="height">The height of the view bounds.</param>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]: The param descriptions here should match the ones in the other constructor.

File: MonoGame.Framework/Graphics/Viewport.cs [hunk 5]
```diff
@@ -154,15 +138,32 @@ public Rectangle Bounds
 				height = value.Height;
 			}
 		}
-		
+
+        /// <summary>
+        /// Gets the actual visible area of this <see cref="Viewport"/> which can differs from <see cref="Bounds"/> on some platforms.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Viewport.cs]:
```
Returns the subset of the viewport that is guaranteed to be visible on a lower quality display.
```

**Comment by Shqrdx** [comment]
@tomspilman Now ok ?... cuz I wanna sleep : p

**Comment by tomspilman** [comment]
@Shqrdx - Don't feel like you need to stay up...  this can always wait. ;)

**Comment by Shqrdx** [comment]
@tomspilman Thank you ! :)

**Pull request merged**